js/ui/control/attribution.js

```diff
@@ -54,9 +54,7 @@ Attribution.prototype = util.inherit(Control, {
             container = this._container = DOM.create('div', className, map.getContainer());
 
         this._update();
-        map.on('source.load', this._update.bind(this));
-        map.on('source.change', this._update.bind(this));
-        map.on('source.remove', this._update.bind(this));
+        map.on('data', this._update.bind(this));
```

Comment:
Add a `dataType === 'source'` guard in `_update` so that it doesn't mutate the DOM more frequently than it did before.